source/ConditionSet.cpp

```diff
@@ -40,15 +42,183 @@ namespace {
+			{"%", [](int64_t a, int64_t b) { return (a % b); }},
+			{"*", [](int64_t a, int64_t b) { return a * b; }},
+			{"+", [](int64_t a, int64_t b) { return a + b; }},
+			{"-", [](int64_t a, int64_t b) { return a - b; }},
+			{"/", [](int64_t a, int64_t b) { return b ? a / b : numeric_limits<int64_t>::max(); }}
 		};
 		
 		auto it = opMap.find(op);
 		return (it != opMap.end() ? it->second : nullptr);
 	}
+	
+	// Indicate if the operation is a comparison or modifies the condition.
+	bool IsComparison(const string &op)
+	{
+		static const set<string> comparison = {
+			"==", "!=", "<", ">", "<=", ">="
+		};
+		return comparison.count(op);
+	}
+	
+	bool IsAssignment(const string &op)
+	{
+		static const set<string> assignment = {
+			"=", "+=", "-=", "*=", "/=", "<?=", ">?="
+		};
+		return assignment.count(op);
+	}
+	
+	bool IsSimple(const string &op)
+	{
+		static const set<string> simple = {
+			"(", ")", "+", "-", "*", "/", "%"
+		};
+		return simple.count(op);
+	}
+	
+	int Precedence(const string &op)
+	{
+		static const map<string, int> precedence = {
+			{"(", 0}, {")", 0},
+			{"+", 1}, {"-", 1},
+			{"*", 2}, {"/", 2}, {"%", 2}
+		};
+		return precedence.at(op);
+	}
+	
+	// Test to determine if unsupported operations are requested.
+	bool HasInvalidOperators(const vector<string> &tokens)
+	{
+		static const set<string> invalids = {
+			"{", "}", "[", "]", "|", "^", "&", "!", "~",
+			"||", "&&", "&=", "|=", "<<", ">>"
+		};
+		for(const string &str : tokens)
+			if(invalids.count(str))
+				return true;
+		return false;
+	}
+	
+	// Ensure the ConditionSet line has balanced parentheses on both sides.
+	bool HasUnbalancedParentheses(const vector<string> &tokens)
+	{
+		int parentheses = 0;
+		for(const string &str : tokens)
+		{
+			if(parentheses < 0)
+				return true;
+			else if(parentheses && (IsAssignment(str) || IsComparison(str)))
+				return true;
+			else if(str == "(")
+				++parentheses;
+			else if(str == ")")
+				--parentheses;
+		}
+		return parentheses;
+	}
+	
+	bool IsValidCondition(const DataNode &node)
+	{
+		const vector<string> &tokens = node.Tokens();
+		int assigns = count_if(tokens.begin(), tokens.end(), IsAssignment);
+		int compares = count_if(tokens.begin(), tokens.end(), IsComparison);
+		if(assigns + compares != 1)
+			node.PrintTrace("An expression must either perform a comparison, or an assign a value:");
+		else if(HasInvalidOperators(tokens))
+			node.PrintTrace("Brackets, braces, exponentiation, and boolean/bitwise math are not supported:");
+		else if(HasUnbalancedParentheses(tokens))
+			node.PrintTrace("Unbalanced parentheses in condition expression:");
+		else if(count_if(tokens.begin(), tokens.end(), [](const string &token)
+				{ return token.size() > 1 && token.front() == '('; }))
+			node.PrintTrace("Parentheses must be separate from tokens:");
+		else
+			return true;
+		
+		return false;
+	}
+	
+	// Converts the given vector of strings into a vector of 64-bit ints.
```

Comment:
This function handles the run-time (as opposed to load-time) conversion of the strings into the values they will have when passed to the associated operators. This is where e.g. `"reputation: Republic"` is converted into e.g. `1034` (or whatever your current reputation with the Republic government happens to be). Rather than interleave reading the condition `map`s with performing the actual condition math, the maps are used, and then the 2 vectors are used.

Any suggestions of a better comment?

Reply:
Something like: Looks up the values of the variables that are stored as strings in the first vector and stores those values in the second vector as 64-bit int values.